cluster/src/main/scala/org/apache/spark/executor/SnappyExecutor.scala

```diff
@@ -89,15 +89,28 @@ class SnappyExecutor(
               s" with dependencies $appDependencies")
           urls = appDependencies.map(name => {
             val localName = name.split("/").last
-            logInfo(s"Fetching file $name for App[$appName]")
-            Utils.fetchFile(name, new File(SparkFiles.getRootDirectory()), conf,
-              env.securityManager, hadoopConf, -1L, useCache = !isLocal)
-            val url = new File(SparkFiles.getRootDirectory(), localName).toURI.toURL
-            Misc.getMemStore.getGlobalCmdRgn.put(ContextJarUtils.functionKeyPrefix + appName, name)
-            url // points to the jar in executor's work directory
+            var fetch = true
+            val firstHyphen = localName.indexOf("-")
+            if (firstHyphen > -1) {
```

Comment:
Just write the logic in comment so that anyone doing a maintenance here can easily understand.